src/com/apeshko/javacore/task_02/FlexibleDynamicArray.java

```diff
@@ -0,0 +1,87 @@
+package com.apeshko.javacore.task_02;
+
+import java.lang.reflect.Array;
+
+public class FlexibleDynamicArray<T> {
+    private T[] items;
+    private int size;
+    private Class<T[]> type;
+
+    public FlexibleDynamicArray(Class<T[]> type, int initialSize) {
+        this.size = 0;
+        this.type = type;
+        this.items = type.cast(Array.newInstance(type.getComponentType(), initialSize));
```

Comment:
you can use Object[] to make it simplier, generic types are object after compilation